worker/uniter/modes.go

```diff
@@ -221,22 +241,30 @@ func modeAbideAliveLoop(u *Uniter) (Mode, error) {
 			return ModeUpgrading(curl), nil
 		case ids := <-u.f.RelationsEvents():
 			creator = newUpdateRelationsOp(ids)
+			userMessage = "updating relation ids"
 		case actionId := <-u.f.ActionEvents():
 			creator = newActionOp(actionId)
+			userMessage = fmt.Sprintf("running action %q", actionId)
 		case tags := <-u.f.StorageEvents():
 			creator = newUpdateStorageOp(tags)
+			userMessage = "updating storage tags"
 		case <-u.f.ConfigEvents():
 			creator = newSimpleRunHookOp(hooks.ConfigChanged)
+			userMessage = "running config-changed hook"
 		case <-u.f.MeterStatusEvents():
 			creator = newSimpleRunHookOp(hooks.MeterStatusChanged)
+			userMessage = "running meter-status-changed hook"
 		case <-collectMetricsSignal:
 			creator = newSimpleRunHookOp(hooks.CollectMetrics)
+			userMessage = "running collect-metrics hook"
 		case hookInfo := <-u.relations.Hooks():
 			creator = newRunHookOp(hookInfo)
+			userMessage = fmt.Sprintf("running %s hook", hookInfo.Kind)
 		case hookInfo := <-u.storage.Hooks():
 			creator = newRunHookOp(hookInfo)
+			userMessage = fmt.Sprintf("running %s hook", hookInfo.Kind)
 		}
```

Comment:
...and it's really only when we hit a `default` case here that we can reasonably say "idle"...

...but we can't reasonably say that, I think, without a timeout for "something interesting" to happen.